src/type/__tests__/definition-test.js

```diff
@@ -342,13 +342,23 @@ describe('Type System: Example', () => {
     ];
     badUnionTypes.forEach(x => {
       expect(() =>
-        new GraphQLUnionType({ name: 'BadUnion', types: [ x ] })
+        new GraphQLUnionType({ name: 'BadUnion', types: [ x ] }).getTypes()
       ).to.throw(
         `BadUnion may only contain Object types, it cannot contain: ${x}.`
       );
     });
   });
 
+  it('allows a thunk for Union\'s types', () => {
+    const types = [ ObjectType ];
+    const union = new GraphQLUnionType({
+      name: 'ThunkUnion',
+      types: () => types
```

Comment:
Isn't the whole point of using a thunk to enable a delay of referencing `ObjectType` until access time?

Shouldn't this be `types: () => [ ObjectType ]` to illustrate that usage?